src/commons_codec/transform/dynamodb.py

```diff
@@ -139,13 +139,13 @@ def values_to_update(self, keys: t.Dict[str, t.Dict[str, str]]) -> str:
         {'humidity': {'N': '84.84'}, 'temperature': {'N': '55.66'}}
 
         OUT:
-        data['humidity] = 84.84, temperature = 55.66
+        data['humidity] = '84.84', temperature = '55.66'
         """
         values_clause = self.deserialize_item(keys)
 
         constraints: t.List[str] = []
         for key_name, key_value in values_clause.items():
-            constraint = f"{self.DATA_COLUMN}['{key_name}'] = {key_value}"
+            constraint = f"{self.DATA_COLUMN}['{key_name}'] = '{key_value}'"
```

Comment:
```suggestion
                        key_value = str(key_value).replace("'", "''")
                        constraint = f"{self.DATA_COLUMN}['{key_name}'] = '{key_value}'"
```

Reply:
We can also check if there is a reusable method in sqlalchemy or crate-python, which would already have sufficient test coverage. And such a method would be easier to re-use, since we will need it several times.

Reply:
cc @amotl @wierdvanderhaar

Reply:
Hi. Thanks for tagging.

We are talking about quoting **values** on this spot, right? We provided a generic quoting function for schema- and table-names per [DatabaseAdapter.quote_relation_name](https://github.com/crate/cratedb-toolkit/blob/v0.0.16/cratedb_toolkit/util/database.py#L43-L79), based on SQLAlchemy's [identifier preparer](https://pydoc.dev/sqlalchemy/latest/sqlalchemy.sql.compiler.IdentifierPreparer.html).

So, Nomen est omen, the identifier preparer is about quoting **identifiers**. I will have a look if there is something similar for **values**.

Reply:
On this spot, is it about proper **quoting**, or rather about **escaping** special characters within the value string at hand?

Reply:
I think that this SO conversation includes good and relevant content and pointers? Please correct me if I'm looking at the wrong thing.

-- https://stackoverflow.com/questions/25104259/how-to-properly-escape-strings-when-manually-building-sql-queries-in-sqlalchemy

Reply:
I will be merging as-is, including @hlcianfagna's suggestion. Generalizing and improving the ad hoc implementation, I am deferring to a subsequent iteration.

- GH-20